xgi/drawing/draw.py

```diff
@@ -1707,111 +1771,154 @@ def draw_dihypergraph(
-    G_aug = _augmented_projection(H_conv)
+        dyad_color = m.to_rgba(dyad_color)
 
-    phantom_nodes = [n for n in list(G_aug.nodes) if n not in list(H_conv.nodes)]
-
-    if pos is None:  # compute positions, first based on original nodes
-        pos_nodes = barycenter_spring_layout(H_conv)
-        # then compute the barycenter positions for the "edge" phantom nodes
-
-        centers = {}
-        for idx, members in DH.edges.members(dtype=dict).items():
-            center = np.zeros((2,))
-            edge_bi_id = phantom_nodes[idx]
-            for node in members:
-                center += pos_nodes[node]
-            centers[edge_bi_id] = center / len(members)
-
-        # merge both pos dictionaries
-        pos_nodes.update(centers)
-        pos = pos_nodes
-
-        # recompute positions from the original positions
-        pos = spring_layout(G_aug, pos=pos, iterations=iterations)
-
-    # draw "node" nodes
     ax, node_collection = draw_nodes(
-        H=H_conv,
-        pos=pos,
+        H=H,
+        pos=node_pos,
         ax=ax,
         node_fc=node_fc,
         node_ec=node_ec,
         node_lw=node_lw,
         node_size=node_size,
-        node_fc_cmap=node_fc_cmap,
         node_shape=node_shape,
-        zorder=4,
-        params=settings,
-        node_labels=node_labels,
+        node_fc_cmap=node_fc_cmap,
+        zorder=2,
+        params=node_settings,
+        node_labels=None,
         rescale_sizes=rescale_sizes,
+        **kwargs,
     )
 
-    # draw "edge" phantom nodes
-    if edge_marker_toggle:
+    if node_labels is not None:
+        draw_node_labels(
+            H,
+            pos=node_pos,
+            node_labels=node_labels,
+            ax_nodes=ax,
+            zorder=3,
+        )
 
-        HG_phantom = Hypergraph()
-        HG_phantom.add_nodes_from(phantom_nodes)  # dummy HG
+    ax, edge_marker_collection = draw_nodes(
+        H=H.dual(),
+        pos=edge_pos,
+        ax=ax,
+        node_fc=edge_marker_fc,
+        node_ec=edge_marker_ec,
+        node_lw=edge_marker_lw,
+        node_size=edge_marker_size,
+        node_shape=edge_marker_shape,
+        node_fc_cmap=edge_marker_fc_cmap,
+        zorder=1,
+        params=edge_marker_settings,
+        node_labels=None,
+        rescale_sizes=rescale_sizes,
+        **kwargs,
+    )
 
-        ax, phantom_node_collection = draw_nodes(
-            H=HG_phantom,
-            pos=pos,
-            ax=ax,
-            node_fc=edge_marker_fc,
-            node_ec=edge_marker_ec,
-            node_lw=edge_marker_lw,
-            node_size=edge_marker_size,
-            node_shape=edge_marker,
-            node_fc_cmap=edge_marker_fc_cmap,
-            zorder=2,
-            params=settings,
-            rescale_sizes=rescale_sizes,
+    if hyperedge_labels is not None:
+        draw_node_labels(
+            H.dual(),
+            pos=edge_pos,
+            node_labels=hyperedge_labels,
+            ax_nodes=ax,
+            zorder=3,
```

Comment:
same question

Reply:
Good catch! You're right, we can just use the `draw_nodes` function.